infrastructure/src/main/java/org/corfudb/infrastructure/FailureHandlerDispatcher.java

```diff
@@ -101,8 +105,30 @@ public void dispatchHandler(IFailureHandlerPolicy failureHandlerPolicy, Layout c
             currentLayout.setRuntime(corfuRuntime);
             sealEpoch(currentLayout);
 
-            // Reconfigure servers if required
-            reconfigureServers(corfuRuntime, currentLayout, newLayout, false);
+            // Mark to-be primary sequencer as not ready if primary sequencer has changed in new
+            // layout.
+            // FIXME:
+            // Since SET_NOT_READY message is epoch UNAWARE, the following scenario can occur:
+            // ManagementServers 1 & 2 are trying to reset sequencer 3. 1 sends SET_NOT_READY but
+            // doesn't receive a response so aborts. 2 sends the SET_NOT_READY, receives response
+            // proceeds and succeeds with reconfiguration. 3 now receives 1's SET_NOT_READY and
+            // accepts it. Who should detect and mark 3 as READY now.
+            if (!currentLayout.getSequencers().get(0).equals(newLayout.getSequencers().get(0))) {
+                // Marking sequencer as not ready.
+                for (String sequencer : newLayout.getSequencers()) {
+                    try {
+                        corfuRuntime.getRouter(sequencer).getClient(SequencerClient.class)
+                                .setNotReady().get();
+                    } catch (InterruptedException | ExecutionException e) {
+                        if (!newLayout.getUnresponsiveServers().contains(sequencer)) {
+                            log.error("Sequencer could not be marked ready: {}", e);
```

Comment:
Typo: please insert "count not be marked ^not^ ready".  Also, this and the other 2/3? new log messages don't follow the recommendations for log message format at https://github.com/CorfuDB/CorfuDB/wiki/Corfu-Style-Guidelines#log-message-style.